### PR TITLE
feat: Hidden fields and metadata for integrations

### PR DESCRIPTION
**Pull request by Dhruwang**
## What does this PR do?

Fixes https://github.com/formbricks/internal/issues/130

Adds hidden fields and metadata to following integrations

- Google sheet
- slack
- Airtable
- Notion

Screenshot 2024-06-11 at 2 57 27 PM
Screenshot 2024-06-11 at 2 57 42 PM
Screenshot 2024-06-11 at 2 58 05 PM
Screenshot 2024-06-11 at 2 57 54 PM

## How should this be tested?

Open integration modal, check `Include Hidden Fields` and `Includes Metadata`

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [x] First PR at Formbricks? [Please sign the CLA!](https://cla-assistant.io/formbricks/formbricks) Without it we wont be able to merge it 🙏

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [x] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

**Comment by jobenjada** [comment]
Hey @Dhruwang 

thanks for your good work, this has been requested several times now :)

Here are the open To Dos:

1. Please also open a PR in the landing page repo with an update to the docs pages, where needed
2. I ran into this error which seems unrelated but maybe you have an idea:

image

It happened when I clickend on "Create custom action" as a survey trigger for an app survey.

🚨 Note: I did not set up all the integrations locally to test it through. The webhook contains the additional information and I suppose testing it on prod once it's live is the more efficient way.

Thanks! :)

File: apps/web/app/api/pipeline/lib/handleIntegrations.ts [hunk 1]
```diff
@@ -43,8 +77,14 @@ const handleAirtableIntegration = async (
...
-        const values = await extractResponses(data, element.questionIds as string[], survey);
-
+        const values = await processDataForIntegration(
+          data,
+          survey,
+          !!element.includeMetadata,
+          !!element.includeHiddenFields,
+          element.questionIds
+        );
+        await airtableWriteData(integration.config.key, element, values);
```
Comment (gupta-piyush19) [inline apps/web/app/api/pipeline/lib/handleIntegrations.ts:87]: can you please remove this? it puts data twice in the airtable.

File: apps/web/app/(app)/environments/[environmentId]/integrations/airtable/components/AddIntegrationModal.tsx [hunk 2]
```diff
@@ -73,12 +75,24 @@ export const AddIntegrationModal = ({
   const [tables, setTables] = useState<TIntegrationAirtableTables["tables"]>([]);
   const [isLoading, setIsLoading] = useState(false);
   const { handleSubmit, control, watch, setValue, reset } = useForm<IntegrationModalInputs>();
+  const [includeHiddenFields, setIncludeHiddenFields] = useState(false);
+  const [includeMetadata, setIncludeMetadata] = useState(false);
```
Comment (gupta-piyush19) [inline apps/web/app/(app)/environments/[environmentId]/integrations/airtable/components/AddIntegrationModal.tsx:79]: can we move these states in the react-hook-form implementation? in other integrations as well?

**Review by gupta-piyush19 — CHANGES_REQUESTED** [review]
Thanks, @Dhruwang for the PR 😊 
functionality works as expected 🚀 
I've added a few small comments, please check them and we'll be able to merge this after that.

File: apps/web/app/(app)/environments/[environmentId]/integrations/airtable/components/AddIntegrationModal.tsx (hunk 2 above)
Comment (mattinannt) [reply apps/web/app/(app)/environments/[environmentId]/integrations/airtable/components/AddIntegrationModal.tsx:79]: Moved into a new ticket https://github.com/formbricks/internal/issues/233

**Comment by gupta-piyush19** [comment]
@Dhruwang thanks for the fixes. LGTM 🚀

**Review by gupta-piyush19 — APPROVED** [review]

**Pull request merged**